synthea/qpp_qicore.yaml

```diff
@@ -0,0 +1,156 @@
+            value: $getField([MedicationRequest.authoredOn])
+  - name: Apply Script​
+    execute_script:
+      - apply_to: resource
+        function_name: addDiagnosis
+        resource_type: Encounter
+        function: |
+          function addDiagnosis(resource, bundle) {
+            const conditionEntry = bundle.entry.find(e => e.resource?.resourceType === 'Condition' && e.resource?.encounter?.reference === `urn:uuid:${resource.id}`);
+            if (conditionEntry){
+              resource.diagnosis = [{
+                condition:{
+                  reference: `Condition/${conditionEntry.resource.id}`
+                }
+              }];
+            }
+          }
+  - name: Create Resources
+    create_resource:
+      - resourceType: ServiceRequest
+        based_on:
+          resource: Procedure
+        fields:
+          - location: ServiceRequest.intent
+            value: order
+          - location: ServiceRequest.encounter.reference
+            value: $getField([Procedure.encounter.reference])
+          - location: ServiceRequest.subject.reference
+            value: $findRef([Patient])
+          - location: ServiceRequest.status
+            value: completed # all procedures are exported as completed
+          - if: Procedure.performed.ofType(Period) # handle value setting for period choice type
+            location: ServiceRequest.authoredOn
+            value: $getField([Procedure.performed.start]) # period choice type
+          - if: Procedure.performed.ofType(dateTime) # handle value setting for datetime choice type
+            location: ServiceRequest.authoredOn
+            value: $getField([Procedure.performed]) #datetime
+          - location: ServiceRequest.code
+            value: $getField([Procedure.code])
+        writeback:
+          - location: Procedure.basedOn.reference
+            value: $setRef([ServiceRequest])
+      - resourceType: DeviceRequest
+        based_on:
+          resource: Device
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-devicerequest
+        fields:
+          - location: DeviceRequest.code.reference
+            value: $findRef([Device])
+          - location: DeviceRequest.authoredOn
+            value: $getField([Device.manufactureDate]) # manufacture time is set 3 weeks before device model's start, so this is close enough
+          - location: DeviceRequest.status
+            value: completed
+          - location: DeviceRequest.intent
+            value: order
+            # Not sure what AdverseEvent would be close to, creating one per patient
+      - resourceType: AdverseEvent
+        based_on:
+          resource: Patient
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-adverseevent
+        fields:
+          - location: AdverseEvent.event.coding
+            value: $randomCode([http://hl7.org/fhir/ValueSet/adverse-event-type])
```

Comment:
We run the risk of creating unrealistic data when we select a random code. For this code set especially, a lot of the codes indicate a death, which may be problematic when the patient doesn't have other data indicating death. We can probably leave it for now and see how it turns out in testing as it may not actually be a big deal. If we want to be extra proactive, we could identify some common bad outcomes that come from synthea modules and create the AdverseEvent resource based on that. (I would guess that Synthea already endeavors to create some adverse events even if they're not labeled as such, and Dylan might have more insight into this.)

Reply:
Sounds good! Or, I am happy to not use `randomCode` here if it would make more sense to just pick a code from that list...but then we wouldn't have as much variability so I am open to either.

Reply:
Off the top of my head I can't think of any adverse events in the modules, other than possibly some of the surgeries have a risk of death. I can look for some though if that's of interest.

This also makes me think there should be some way of saying "random code from these specific few codes" rather than requiring the use of a terminology server and a published ValueSet. I can probably get that implemented pretty quickly